src/Compilers/CSharp/Test/Syntax/Parsing/LineSpanDirectiveParsingTests.cs

```diff
@@ -1915,5 +1915,201 @@ public void NotUTF8StringLiteral_04()
+                );
+
+            N(SyntaxKind.LineDirectiveTrivia);
+            {
+                N(SyntaxKind.HashToken);
+                N(SyntaxKind.LineKeyword);
+                N(SyntaxKind.NumericLiteralToken, "1");
+                N(SyntaxKind.StringLiteralToken, "\"\"\"" + @"
+file.cs
+" + "\"\"\"");
+                N(SyntaxKind.EndOfDirectiveToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void NotUTF8StringLiteral_10()
+        {
+            string source = @"#line 1 @""""""
+file.cs
+""""""u8";
+
+            UsingLineDirective(source, options: null,
+                // (1,9): error CS1578: Quoted file name, single-line comment or end-of-line expected
+                // #line 1 @"""
+                Diagnostic(ErrorCode.ERR_MissingPPFile, "@").WithLocation(1, 9)
+                );
+
+            N(SyntaxKind.LineDirectiveTrivia);
+            {
+                N(SyntaxKind.HashToken);
+                N(SyntaxKind.LineKeyword);
+                N(SyntaxKind.NumericLiteralToken, "1");
+                N(SyntaxKind.EndOfDirectiveToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void NotUTF8StringLiteral_11()
+        {
+            string source = @"#line 1 """"""
+file.cs
+""""""U8";
+
+            UsingLineDirective(source, options: null,
+                // (1,9): error CS8996: Raw string literals are not allowed in preprocessor directives.
+                // #line 1 """
+                Diagnostic(ErrorCode.ERR_RawStringNotInDirectives, "").WithLocation(1, 9),
+                // (2,4): error CS1025: Single-line comment or end-of-line expected
+                // """U8
+                Diagnostic(ErrorCode.ERR_EndOfPPLineExpected, "U8").WithLocation(2, 4)
+                );
+
+            N(SyntaxKind.LineDirectiveTrivia);
+            {
+                N(SyntaxKind.HashToken);
+                N(SyntaxKind.LineKeyword);
+                N(SyntaxKind.NumericLiteralToken, "1");
+                N(SyntaxKind.StringLiteralToken, "\"\"\"" + @"
+file.cs
+" + "\"\"\"");
+                N(SyntaxKind.EndOfDirectiveToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void NotUTF8StringLiteral_12()
+        {
+            string source = @"#line 1 @""""""
+file.cs
+""""""U8";
+
+            UsingLineDirective(source, options: null,
+                // (1,9): error CS1578: Quoted file name, single-line comment or end-of-line expected
+                // #line 1 @"""
+                Diagnostic(ErrorCode.ERR_MissingPPFile, "@").WithLocation(1, 9)
+                );
+
+            N(SyntaxKind.LineDirectiveTrivia);
+            {
+                N(SyntaxKind.HashToken);
+                N(SyntaxKind.LineKeyword);
+                N(SyntaxKind.NumericLiteralToken, "1");
+                N(SyntaxKind.EndOfDirectiveToken);
+            }
+            EOF();
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Please add tests for enhanced line directives with UTF8 string literals and UTF8 raw string literals:
```csharp
#line (1, 2)-(3, 4) ""file.cs""u8
#line (1, 2)-(3, 4) """file.cs"""u8
```

Reply:
> Please add tests for enhanced line directives with UTF8 string literals and UTF8 raw string literals:

If you insist, I can add the suggested test. However, I do not see much benefits in testing every single directive flavor. The goal is to test how we scan literals in context of directives, i.e. to confirm that we do not recognize the suffix.

Please let me know if you still want me to add the tests.